test/parallel/test-child-process-execfile.js

```diff
@@ -47,3 +47,42 @@ const execOpts = { encoding: 'utf8', shell: true };
     assert.ifError(err);
   }));
 }
+
+{
+  // Verify the shell option is only passed explicitly
+  // in the options object and not inherited from the
+  // object prototype
+  Object.prototype.shell = true;
+
+  execFile(
+    'date',
+    ['; echo baz'],
+    common.mustCall((e, stdout) => {
+      assert.notStrictEqual(e, null);
+      assert.strictEqual(stdout.trim().indexOf('baz'), -1);
+    })
+  );
+
+  Reflect.deleteProperty(Object.prototype, 'shell');
+}
+
+{
+  // Verify the shell option is only passed explicitly
+  // in the options object and not inherited from the
+  // object prototype
+  Object.prototype.shell = true;
+
+  let assertionExpected = 0;
+  try {
+    execFileSync(
+      'date',
+      ['; echo baz']
+    );
+  } catch (err) {
+    assertionExpected++;
+    assert.strictEqual(err.status, 1);
+  }
+
+  assert.strictEqual(assertionExpected, 1);
```

Comment:
Could you use `assert.throws()` here?

Reply:
Yep good idea! I'll update.

Reply:
@cjihrig I changed the test code to use `assert.throws`, can you see if I'm using that how you expect me to?